src/elysium/sigmawallet.h

```diff
@@ -67,14 +67,55 @@ class SigmaWallet
         >
     > MintPool;
 
+protected:
+    class WalletDB {
+    public:
+        WalletDB(std::string const &walletFile);
+
+    public:
+        virtual bool WriteMint(SigmaMintId const &id, SigmaMint const &mint, CWalletDB *db = nullptr) = 0;
+        virtual bool ReadMint(SigmaMintId const &id, SigmaMint &mint, CWalletDB *db = nullptr) const = 0;
+        virtual bool EraseMint(SigmaMintId const &id, CWalletDB *db = nullptr) = 0;
+        virtual bool HasMint(SigmaMintId const &id, CWalletDB *db = nullptr) const = 0;
+
+        virtual bool WriteMintId(uint160 const &hash, SigmaMintId const &mintId, CWalletDB *db = nullptr) = 0;
+        virtual bool ReadMintId(uint160 const &hash, SigmaMintId &mintId, CWalletDB *db = nullptr) const = 0;
+        virtual bool EraseMintId(uint160 const &hash, CWalletDB *db = nullptr) = 0;
+        virtual bool HasMintId(uint160 const &hash, CWalletDB *db = nullptr) const = 0;
+
+        virtual bool WriteMintPool(std::vector<MintPoolEntry> const &mints, CWalletDB *db = nullptr) = 0;
+        virtual bool ReadMintPool(std::vector<MintPoolEntry> &mints, CWalletDB *db = nullptr) = 0;
+
+        virtual void ListMints(std::function<void(SigmaMintId&, SigmaMint&)> const&, CWalletDB *db = nullptr) = 0;
+
+    protected:
+        // Helper
+        struct DBDeleter {
+        private:
+            bool mustDelete;
+
+        public:
+            DBDeleter(bool mustDelete);
+
+        public:
+            void operator()(CWalletDB* db);
+        };
+
+        std::unique_ptr<CWalletDB, DBDeleter> EnsureDBConnection(CWalletDB *db) const;
```

Comment:
After a discussion we should create a new class to wrap a `CWalletDB` instance to be able to select a user privided or the default one.